lib/blockchain_double.js

```diff
@@ -466,22 +478,10 @@ BlockchainDouble.prototype.sortByPriceAndNonce = function() {
 
 BlockchainDouble.prototype.processCall = function(tx, blockNumber, callback) {
   var self = this;
-  var stateManager = this.vm.stateManager;
-
-  var cleanUpAndReturn = function(err, result, startingStateRoot) {
-    stateManager.revert(function(e) {
-      // For defaultBlock, undo state root changes
-      if (startingStateRoot) {
-        stateManager.setStateRoot(startingStateRoot, () => {
-          callback(err || e, result);
-        });
-      } else {
-        callback(err || e, result);
-      }
-    });
-  };
+  var vm = this.createVMFromStateTrie(this.stateTrie.copy());
```

Comment:
`processCall` now uses an ephemeral copy of the vm based on the current state trie (it ignores the cache since things in the cache aren't guaranteed to be written to the trie, especially in an `asyncRequestProcessing` context).